File: CompilerRR/CompilerRR/src/ch/fhnw/cpib/compiler/classes/GlobalImport.java
package ch.fhnw.cpib.compiler.classes;

import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IOptionalCHANGEMODE;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IOptionalFLOWMODE;
import ch.fhnw.cpib.compiler.scanner.Token;

public class GlobalImport implements IConcSyn.IGlobalImport {

	IConcSyn.IOptionalFLOWMODE flowmode;
	IConcSyn.IOptionalCHANGEMODE changemode;
	Token ident;
	public GlobalImport(IOptionalFLOWMODE flowmode, IOptionalCHANGEMODE changemode, Token ident) {
		super();
		this.flowmode = flowmode;
		this.changemode = changemode;
		this.ident = ident;
	}
	
	
}

File: CompilerMR/IML_Compiler/IMLParser/DataTypes/Concrete/GlobalImport.cs
using IML_Compiler.IMLParser.DataTypes.Abstract;
using IML_Compiler.IMLScanner.DataTypes;

namespace IML_Compiler.IMLParser.DataTypes.Concrete
{
    class GlobalImport : IGlobalImport
    {
        public IOptionalFlowMode OptionalFlowMode { get; private set; }
        public IOptionalChangeMode OptionalChangeMode { get; private set; }
        public Ident Ident { get; private set; }

        public GlobalImport (IOptionalFlowMode flowMode, IOptionalChangeMode changeMode,
            Ident ident)
        {
            OptionalFlowMode = flowMode;
            OptionalChangeMode = changeMode;
            Ident = ident;
        }

        public AbsGlobalImport ToAbySyn()
        {
            return new AbsGlobalImport()
            {
                ChangeMode = OptionalChangeMode.ToAbySyn(),
                FlowMode = OptionalFlowMode.ToAbsSyn(),
                Ident = this.Ident.value
            };
        }
    }
}
